src/test/data/tx_invalid.json

```diff
@@ -174,7 +174,7 @@
 "0100000001000100000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000ff64cd1d", "P2SH,CHECKLOCKTIMEVERIFY"],
 
 ["Argument 2^32 with nLockTime=2^32-1"],
-[[["0000000000000000000000000000000000000000000000000000000000000100", 0, "4294967296 CHECKLOCKTIMEVERIFY 1"]],
```

Comment:
Shouldn't this version be kept as an example of an *_invalid* script?

Reply:
OK, thanks. This is a good idea I think, I was hesitant to add more tests. If I understood you correctly, the tests are invalid here, so I guess you mean adding something like:

```
["Argument 2^32 (decimal value hence clamped to 0xffffffff) with nLockTime=2^32-1"],
[[["0000000000000000000000000000000000000000000000000000000000000100", 0, "4294967296 CHECKLOCKTIMEVERIFY 1"]],
"0100000001000100000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000fffffffe", "P2SH,CHECKLOCKTIMEVERIFY"],
```

If this is correct, I guess I should add similar tests for other test cases too ?

Reply:
Feel free to *add* as many tests as you like, as long as all existing proper test cases also stay there

Reply:
OK great, I added 5 more tests in the branch.

Reply:
What I mean is that this shouldn't be silently clamped. The test here looks right. Why would it be safe to parse a script and silently change it while parsing?

Reply:
OK, so you are suggesting in `core_read.cpp` to return an `std::runtime_error` when the `atoi64` is beyond the the range -0xffffffff..0xffffffff instead ?

Reply:
Yes, some kind of failure

Reply:
OK, thanks, this is clearly a possibility. 

Note that for now, no test trigger the already existing `std::runtime_error`. `script_json_test` doesn't handle exceptions, so it should be modified to catch them (maybe in a different PR then ?). Testcases will have to be modified to handle the exception too I guess. 

(the fuzzer harness already handle exceptions well)

I'm not sure which possibility (clamping or throwing an error) fits more the style of the codebase. Maybe getting an ACK from @sipa for the exception first ?

Reply:
@practicalswift hm sorry but I guess it's better to ask first:  if you return an `std::runtime_error`, how to describe the error in the JSON test file ?  

I guess that adding a new value to the `ScriptError_t` enum is too intrusive. 
Adding a special `if/else` case in `script_test.cpp:ParseScriptError` seems too hackish.
Reusing an existing `ScriptError_t` seems the less odd option - something like `SCRIPT_ERR_BAD_OPCODE` or `SCRIPT_ERR_OP_COUNT` ?

Reply:
I don't think there is a need. (tx/script)_(valid/invalid) are testing the transaction and script validity logic - not the script parser.

If a particular test now causes an exception to be thrown, just write that test differently. It's still testing the same thing.

Reply:
OK, thanks, so if I understand correctly this means:
- the tests above with `99999..999` should be removed since they just test the script parser
- since we still want the test containing `4294967296 CHECKLOCKTIMEVERIFY 1` and throw an error while parsing it (the reason of this code review), consider that if `ParseScript` in `tx_invalid` throws an error, the test is valid ?

Reply:
OK, so as stated below, if a test contains "4294967296 CHECKLOCKTIMEVERIFY 1" or "999...999", this is a parse error which should be covered in `bitcoin-util-test.json`, not here. I believe this resolve this issue ?